blueocean-commons/src/main/java/io/jenkins/blueocean/commons/BlueUrlTokenizer.java

```diff
@@ -271,6 +272,75 @@ public boolean lastPartIs(@Nonnull UrlPart urlPart, @Nonnull String value) {
         return false;
     }
 
+    public static String reconstructJobNamePath(String name, String fullName) {
+        return reconstructJobNamePath(name, fullName, "/");
+    }
+
+    public static String reconstructJobNamePath(String name, String fullName, String separator){
+        StringBuilder pipelinePath = new StringBuilder();
+        String fullPath;
+        List<String> pathTokens;
+
+        //
+        // The Job "fullName" is typically suffixed with the "name".
+        // The name is NOT part of the path to the job, so lets remove
+        // it from the fullName before we parse and reconstruct the path.
+        // This, along with reencoding of the name, helps us to eliminate
+        // the problems where a Job is programatically created with a name
+        // that's not URL encoded and contains characters that should be
+        // encoded e.g. slashes (see JENKINS-41425).
+        //
+        if (fullName.endsWith(name)) {
```

Comment:
How does it distinguish legal '/' from the one present in the job name. That is a job nested inside folders and folders are created with illegal chars. For example job name 'p/p1/ inside folder 'f/f1' should be encoded as full name  `f%2Ff1/p%2Fp1`.